logger/logger.go

```diff
@@ -3,54 +3,27 @@ package logger
 
 import (
 	"fmt"
-	"sync"
 )
 
+type Fields map[string]interface{}
+
 // Logger is a generic logging interface
 type Logger interface {
 	// Init initialises options
 	Init(options ...Option) error
 	// Level returns the logging level
 	Level() Level
-	// Log inserts a log entry.  Arguments may be handled in the manner
-	// of fmt.Print, but the underlying logger may also decide to handle
-	// them differently.
-	Log(level Level, v ...interface{})
-	// Logf insets a log entry.  Arguments are handled in the manner of
-	// fmt.Printf.
-	Logf(level Level, format string, v ...interface{})
-	// Fields set fields to always be logged
-	Fields(fields map[string]interface{}) Logger
-	// Error set `error` field to be logged
-	Error(err error) Logger
 	// SetLevel updates the logging level.
 	SetLevel(Level)
 	// String returns the name of logger
 	String() string
+	// log at given level with message, fmtArgs and context fields
+	Log(level Level, template string, fmtArgs []interface{}, fields Fields)
```

Comment:
Is there a particular reason we moved away from the old interface? This is really not as clean as the previous implementation was.